sample_factory/algorithms/appo/shared_buffers.py

```diff
@@ -205,10 +206,22 @@ def set_data_func(self, x, index, new_data):
         if isinstance(new_data, (dict, TensorDict)):
             for new_data_key, new_data_value in new_data.items():
                 self.set_data_func(x[new_data_key], index, new_data_value)
-        elif isinstance(new_data, torch.Tensor):
-            x[index].copy_(new_data)
-        elif isinstance(new_data, np.ndarray):
-            t = torch.from_numpy(new_data)
+        if torch.is_tensor(x):
```

Comment:
So before the change TensorDict could only be a dict of Pytorch tensors, and now it can be also a dict of numpy arrays. Clever